extensions/kusto/src/utils.ts

```diff
@@ -0,0 +1,295 @@
+
+	return title;
+}
+
+export function removeOldLogFiles(logPath: string, prefix: string): JSON {
+	return findRemoveSync(logPath, { age: { seconds: getConfigLogRetentionSeconds() }, limit: getConfigLogFilesRemovalLimit() });
+}
+
+export function getConfiguration(config: string = extensionConfigSectionName): vscode.WorkspaceConfiguration {
+	return vscode.workspace.getConfiguration(extensionConfigSectionName);
+}
+
+export function getConfigLogFilesRemovalLimit(): number {
+	let config = getConfiguration();
+	if (config && config[configLogFilesRemovalLimit]) {
+		return Number((config[configLogFilesRemovalLimit]).toFixed(0));
+	}
+	else {
+		return undefined;
+	}
+}
+
+export function getConfigLogRetentionSeconds(): number {
+	let config = getConfiguration();
+	if (config && config[configLogRetentionMinutes]) {
+		return Number((config[configLogRetentionMinutes] * 60).toFixed(0));
+	}
+	else {
+		return undefined;
+	}
+}
+
+export function getConfigTracingLevel(): string {
+	let config = getConfiguration();
+	if (config) {
+		return config[configTracingLevel];
+	}
+	else {
+		return undefined;
+	}
+}
+
+export function getLogFileName(prefix: string, pid: number): string {
+	return `${prefix}_${pid}.log`;
+}
+
+export function getCommonLaunchArgsAndCleanupOldLogFiles(logPath: string, fileName: string, executablePath: string): string[] {
+	let launchArgs = [];
+	launchArgs.push('--log-file');
+	let logFile = path.join(logPath, fileName);
+	launchArgs.push(logFile);
+
+	console.log(`logFile for ${path.basename(executablePath)} is ${logFile}`);
+	console.log(`This process (ui Extenstion Host) is pid: ${process.pid}`);
+	// Delete old log files
+	let deletedLogFiles = removeOldLogFiles(logPath, fileName);
+	console.log(`Old log files deletion report: ${JSON.stringify(deletedLogFiles)}`);
+	launchArgs.push('--tracing-level');
+	launchArgs.push(getConfigTracingLevel());
+	return launchArgs;
+}
+
+export function ensure(target: object, key: string): any {
+	if (target[key] === void 0) {
+		target[key] = {} as any;
+	}
+	return target[key];
+}
+
+export interface IPackageInfo {
+	name: string;
+	version: string;
+	aiKey: string;
+}
+
+export function getPackageInfo(packageJson: any): IPackageInfo {
+	if (packageJson) {
+		return {
+			name: packageJson.name,
+			version: packageJson.version,
+			aiKey: packageJson.aiKey
+		};
+	}
+	return undefined;
+}
+
+export function generateUserId(): Promise<string> {
+	return new Promise<string>(resolve => {
+		try {
+			let interfaces = os.networkInterfaces();
+			let mac;
+			for (let key of Object.keys(interfaces)) {
+				let item = interfaces[key][0];
+				if (!item.internal) {
+					mac = item.mac;
+					break;
+				}
+			}
+			if (mac) {
+				resolve(crypto.createHash('sha256').update(mac + os.homedir(), 'utf8').digest('hex'));
+			} else {
+				resolve(generateGuid());
+			}
+		} catch (err) {
+			resolve(generateGuid()); // fallback
+		}
+	});
+}
+
+export function generateGuid(): string {
```

Comment:
Don't generate UUIDs on your own - it's very likely to make a mistake

Reply:
Removed it.